src/software/world/robot_state.h

```diff
@@ -56,6 +56,48 @@ class RobotState
      */
     AngularVelocity angularVelocity() const;
 
+    /**
+     * Returns if the ball is in the break beam
+     *
+     * @return ball in break beam
+     */
+    bool ballInBeam() const;
+
+    /**
+     * Update ball in break beam
+     *
+     * @param ball_in_beam whether ball is in beam
+     */
+    void updateBallInBeam(bool ball_in_beam);
+
+    /**
+     * Returns time in milliseconds since last chip
+     *
+     * @return ms since last chip
+     */
+    unsigned long timeSinceLastChip() const;
+
+    /**
+     * Update time in milliseconds since last chip
+     *
+     * @param time ms since last chip
+     */
+    void updateTimeSinceLastChip(unsigned long time);
+
+    /**
+     * Returns time in milliseconds since last kick
+     *
+     * @return ms since last kick
+     */
+    unsigned long timeSinceLastKick() const;
+
+    /**
+     * Update time in milliseconds since last kick
+     *
+     * @param time ms since last kick
+     */
+    void updateTimeSinceLastKick(unsigned long time);
```

Comment:
`update` -> `set` for consistency with the rest of the codebase

Reply:
done